lib/renderer/window-setup.ts

```diff
@@ -230,28 +245,31 @@ export const windowSetup = (
     });
   }
 
-  if (!process.sandboxed && !rendererProcessReuseEnabled) {
+  if (!process.sandboxed && !rendererProcessReuseEnabled && !(inMainWorld && contextIsolationEnabled)) {
```

Comment:
this conditional is super confusing, perhaps we can do de Morgan + naming to it?
```suggestion
  const usesNativeHistory = process.sandboxed || rendererProcessReuseEnabled;
  const inPrivilegedContext = !contextIsolationEnabled || !inMainWorld;
  if (!usesNativeHistory && inPrivilegedContext) {
```